actions/restore.py

```diff
@@ -190,11 +191,20 @@ def probe_forked_etcd():
 def reconfigure_client_advertise():
     ''' Reconfigure the backup to use host network addresses for client advertise
         instead of the assumed localhost addressing '''
-    cmd = "/snap/bin/etcd.etcdctl member list"
-    members = check_output(split(cmd))
-    member_id = members.split(b':')[0].decode('utf-8')
 
-    raw_update = "/snap/bin/etcd.etcdctl member update {0} http://{1}:{2}"
+    loop = 0
+    MAX_WAIT = 10
+
+    while loop < MAX_WAIT:
+        try:
+            cmd = "/snap/bin/etcd.etcdctl member list"
+            members = check_output(split(cmd))
+            member_id = members.split(b',')[0].decode('utf-8')
+            break
+        except CalledProcessError:
+            loop = loop + 1
```

Comment:
could be nice to log here that we're going for another loop and perhaps even wait for a bit before retrying

Reply:
What about splitting the retry or retry & wait logic in a separate function/decorator? It will be easier to unit test too.

There are library that offer this kind of utils, like [retry](https://pypi.org/project/retry/) or [backoff](https://pypi.org/project/backoff/).

Reply:
or [tenacity](https://tenacity.readthedocs.io/en/latest/)

the trouble with libraries(like these, but not necessarily) is making sure they have good back support for releases like xenial (yes, we still support python 3.6)

Reply:
As written, when `loop` == `MAX_WAIT`, we'll gobble up the exception and fall down to `member update` later, which will presumably fail since `member list` has failed.

Perhaps log the event as WARNING until our final loop, then log as ERROR and `raise`.